packages/ui-core/src/components/dynamic-portal/dynamic-portal-component.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect, useRef } from 'react';
+import { createPortal } from 'react-dom';
+import { UI_LAYERS } from '../../constants';
+
+type Props = {
+  children: React.ReactNode;
+  zIndex?: number;
+};
+
+export const DynamicPortal = ({
+  children,
+  zIndex,
+}: Props): React.ReactPortal => {
+  const el = useRef(null);
+  useEffect(() => {
+    el.current = document.createElement('div');
+    el.current.id = 'dynamic-portal';
+    el.current.style.zIndex = `${zIndex || UI_LAYERS.tooltip}`;
+    el.current.setAttribute('data-testid', 'dynamic-portal');
+    document.body.appendChild(el.current);
+    return () => {
+      document.body.removeChild(el.current);
+    };
+  }, [zIndex]);
```

Comment:
why it's dependent on `zIndex` that is optional?

Reply:
If we pass zIndex as a prop React will simply skip that effect after initial render.
But it will not trigger that effect on prop change if we don’t pass zIndex inside dependency array.